docs/fxa-engineering/fxa-sub-platform.md

```diff
@@ -169,13 +169,108 @@ server:
+
+## PayPal Integration
+
+PayPal can be configured as an additional payment provider in the Subscription Platform.  PayPal's [Express Checkout Reference Transactions](https://developer.paypal.com/docs/archive/express-checkout/integration-guide/ECReferenceTxns/) is the feature that enables the Subscription Platform to use PayPal as a payment provider for recurring subscriptions.  The customer's PayPal Billing Agreement ID is saved for future subscription invoices.  See [the diagram below](#paypal-checkout) for details on this process.
+
+The PayPal paid subscriptions are still driven by Stripe's subscription and invoicing model.  The key difference is that PayPal paid subscriptions have a [collection method](https://stripe.com/docs/api/subscriptions/object#subscription_object-collection_method) of `send_invoice`.  The [PayPal processor](#paypal-processor) is used to pay the inovices with the customer's billing agreement ID.
+
+### PayPal Accounts
+
+For development, you will need a PayPal developer account.  Once you are in the PayPal developer dashboard, navigate to "Accounts" under the Sandbox section of the menu.  Here you should create a pair of personal and business sandbox accounts.  The personal account will be use to test as the customer; the business account will be receiving the subscription payments.
+
+Once you've added the account pair, navigate to the bussines account by selecting "View/edit account".  Now click on the "API Credentials" tab.  You'll need the "NVP/SOAP Sandbox API Credentials" for the next section.
+
+### Configuration
+
+#### Auth Server
+
+In order to enable and use PayPal in the auth server, set the following configuration options
+
+```
+{
+  "subscriptions": {
+    "paypalNvpSigCredentials": true,
+    "sandbox": true,
+    "user": "your PayPal NVP API User name",
+    "pwd": "your PayPal NVP API password",
+    "signature": "your PayPal NVP API signature"
+  }
+}
+```
+
+The environment variables equivalent would be
+```
+SUBSCRIPTIONS_PAYPAL_ENABLED=true \
+PAYPAL_SANDBOX=true \
+PAYPAL_NVP_USER='your PayPal NVP API User name' \
+PAYPAL_NVP_PWD='your PayPal NVP API password' \
+PAYPAL_NVP_SIGNATURE='your PayPal NVP API signature'
+```
+
+#### Payments Server
+
+The Payments frontend also does not offer PayPal as payment provider by default.  To enable the feature, set the following configuration options
+
+```
+{
+  "featureFlags": {
+    "usePaypalUIByDefault": true,
+    }
+  },
+  "paypal": {
+    "clientId": "sb",
+    "apiUrl": "https://www.sandbox.paypal.com",
+    "scriptUrl": "https://www.paypal.com"
+  }
+}
+```
+
+Or use the environment variables
+
+```
+FEATURE_USE_SCA_PAYMENT_UI_BY_DEFAULT=true \
+PAYPAL_CLIENT_ID='sb' \
+PAYPAL_API_URL='https://www.sandbox.paypal.com' \
+PAYPAL_SCRIPT_URL='https://www.paypal.com'
+```
+
+The paypal*/PAYPAL_* values are the defaults in the repo.  For local development, you do not need to change them.
```

Comment:
You could also mention (perhaps in a subsection here) that we use the [PayPal JavaScript SDK](https://developer.paypal.com/docs/business/checkout/set-up-standard-payments/) on the front-end with the NVP/SOAP APIs (the SDK defaults to REST APIs).

Reply:
Would https://developer.paypal.com/docs/business/checkout/configure-payments/single-page-app/ be a better link since that's what we are doing?

Reply:
Added the The PayPal button section below.